### DISCUSSION TITLE
Is this possible to create custom design component and use it in field  ?

### DISCUSSION THREAD

**Discussion by 7Hemant** [Thinkmill/keystatic#595] [Q&A] [answered]
Hey ! 

I'm trying to create my own components, like a button component, using **import { component } from "@keystatic/core"**;. However, I'm facing problems when trying to edit the button field. I haven't found any official documents on it.

**Comment by JedWatson** [comment, marked as answer]
I think you mean in the `document` editor?

I've just cleaned up the examples in the `docs` website in the Keystatic repo here: https://github.com/Thinkmill/keystatic/tree/main/docs/src/component-blocks

We'll be updating the docs soon with better instructions on how to do this based on these examples but reading the code for the docs site may point you in the right direction

**Reply by 7Hemant** [reply]
Thanks ,this example is useful for me.
